### ISSUE TITLE
Introduce "auto-switch" next to "auto-reload"

### ISSUE THREAD

**Issue by ETLegacyTeam** [etlegacy/etlegacy#388]
Author: @realkemon (keMoN)
Date: 2013-11-06
Redmine Issue: 388, https://dev.etlegacy.com/issues/388

---

Allow players to modify the gun-behavior when the clip empties.
Right now you can either switch "auto-reload" on or off. My suggestion is to add "auto-switch" to that selection.
This would directly switch from primary to secondary and the other way around when the magazine gets empty. IF both weapons are empty it would switch to knife!

on IRC discussed problematics:
-changing of gameplay 
--right now auto-reload on/off doesn't create any noticable advantage, why should auto-switch?

**Comment by ETLegacyTeam** [comment]
From: @rmarquis (Spyhawk)
Date: 2013-11-06T17:20:20Z

---

> –right now auto-reload on/off doesn’t create any noticable advantage, why should auto-switch?

Just some little thought: actually, auto-reload creates a disavantage, since you'd better switch to a loaded weapon in fight.

**Comment by ETLegacyTeam** [comment]
From: @realkemon (keMoN)
Date: 2013-11-06T17:28:00Z

---

I totally agree with you and by introducing this feature players can actually empty the gun instead of switching to the secondary weapon when it still contains 3-4 bullets.

**Comment by ETLegacyTeam** [comment]
From: @realkemon (keMoN)
Date: 2013-11-06T18:44:24Z

---

since there occured certain misunderstandings in IRC:
I'm talking ONLY about the situation when your clip is empty, but you still have other clips left.
Some prefer the comfort to use *auto-reload* which is more comfortably (depending on your preferies/style of play) but a disadvantage in certain situations.
*auto-switch* would be the exact same thing, simply from a different angle. It brings you comfort (when you switch in-fight anyways) but raises disadvantages in other situations.

Basically you could specify the gun-behavior when the clip is empty with these parameters: OFF/AUTO-RELOAD/AUTO-SWITCH

Why should we provide the possibility to make the game more comfortable for some players, but not for others?
Imo ET is the most tweakable game there is. You can modify everything from gameplay to graphics, why not increase the personalization?

**Comment by ETLegacyTeam** [comment]
From: @adanski (Dragonji)
Date: 2013-12-31T16:55:46Z

---

Sounds good.
